client/lib/parsing.rs

```diff
@@ -965,6 +966,119 @@ mod tests {
         );
     }
 
+    #[test]
+    fn should_parse_valid_deploy_params() {
+        // create secret key file in tempdir.
+        let keys_dir = tempdir().expect("Failed to create temp dir.");
+        let secret_key_path = keys_dir.path().join("key.pem");
+        let secret_key_path_clone = secret_key_path.clone();
+        let secret_key_path_str = secret_key_path_clone.to_str().unwrap();
```

Comment:
Maybe merge this into one line:
`let secret_key_path_str = secret_key_path.clone().to_str()?;`

Reply:
Could you try `let secret_key_path_str = secret_key_path.to_str().unwrap()` directly, see if the compiler complains about the value being moved.